spring-cloud-services-spring-connector/src/main/java/io/pivotal/spring/cloud/service/eureka/EurekaInstanceAutoConfiguration.java

```diff
@@ -69,6 +69,12 @@
 	@Value("${spring.cloud.services.registrationMethod:route}")
 	private String registrationMethod;
 
+	@Value("${virtual-host-name:#{null}}")
+	private String providedVirtualHostname;
+
+	@Value("${secure-virtual-host-name:#{null}}")
+	private String providedSecureVirtualHostname;
+
```

Comment:
It's a fair cop governor! I would like to avoid field injection too. @royclarkson has kindly agreed to raise a separate story to avoid scope creep here.